File: 2015/AI/AIClient_Python/src/event/UpdateBoxEvent.py
'''
Created on Jan 11, 2015

@author: scarriere
'''
from event.IngoingEvent import IngoingEvent
from aiclient.Singleton import Singleton
from world.World import World

class UpdateBoxEvent(IngoingEvent):

    def __init__(self):
        self.x = 0
        self.y = 0
    
    def fillArguments(self, string):
        arguments = string.split(":", 2)
        
        self.x = int(arguments[0])
        self.y = int(arguments[1])
        
    def execute(self):
        world = Singleton(World)
        world._updateBox(self.x, self.y)
        

File: 2015/AI/AIClient_JAVA/src/event/UpdateBoxEvent.java
/* ------------------------------------------------------------------------------
 ** _________   _________      ________    _____      _____  ___________ _________
 ** \_   ___ \ /   _____/     /  _____/   /  _  \    /     \ \_   _____//   _____/
 ** /    \  \/ \_____  \     /   \  ___  /  /_\  \  /  \ /  \ |    __)_ \_____  \ 
 ** \     \____/        \    \    \_\  \/    |    \/    Y    \|        \/        \
 **  \______  /_______  /     \______  /\____|__  /\____|__  /_______  /_______  /
 **         \/        \/             \/         \/         \/        \/        \/ 
 **
 ** UpdateBoxEvent.java
 ** Event that send a box position
 **
 ** Author: Samuel-Ricardo Carriere
 ** ------------------------------------------------------------------------------*/

package event;

import java.awt.Point;

import world.World;

public class UpdateBoxEvent extends IngoingEvent {
	private Point position = new Point();

	@Override
	public void execute() {
		World.getInstance().updateBox(position);
	}

	@Override
	public boolean fillArguments(String message) {
		String[] messageParts = message.split(SEPARATOR, 3);

		position.x = convertCharToNumeral(messageParts[0]);
		position.y = convertCharToNumeral(messageParts[1]);
		return true;
	}

}
